streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -40,9 +43,27 @@
  * A join is symmetric in the sense, that a join specification on the first stream returns the same result record as
  * a join specification on the second stream with flipped before and after values.
  * <p>
- * Both values (before and after) must not result in an "inverse" window,
- * i.e., lower-interval-bound must not be larger than upper-interval.bound.
+ * Both values (before and after) must not result in an "inverse" window, i.e., upper-interval-bound cannot be smaller
+ * than lower-interval bound.
```

Comment:
do you want this to be "lower-interval-bound" to be consistent with "upper-interval-bound", or vice-versa?